radix-transactions/src/manifest/static_resource_movements/effect.rs

```diff
@@ -112,84 +117,67 @@ impl StaticInvocationResourcesOutput for TypedNativeInvocation {
-                                input.output(node_id, inputs, instruction_index)
+                                input.output(details)
                             }
                             AccountMethod::TryDepositOrAbort(input) => {
-                                input.output(node_id, inputs, instruction_index)
+                                input.output(details)
                             }
-                            AccountMethod::Burn(input) => input.output(node_id, inputs, instruction_index),
+                            AccountMethod::Burn(input) => input.output(details),
                             AccountMethod::BurnNonFungibles(input) => {
-                                input.output(node_id, inputs, instruction_index)
+                                input.output(details)
                             }
                             AccountMethod::AddAuthorizedDepositor(input) => {
-                                input.output(node_id, inputs, instruction_index)
+                                input.output(details)
                             }
                             AccountMethod::RemoveAuthorizedDepositor(input) => {
-                                input.output(node_id, inputs, instruction_index)
+                                input.output(details)
                             }
                             AccountMethod::TryDepositOrRefund(input) => {
-                                input.output(node_id, inputs, instruction_index)
-                            }
-                            // TODO: A bit of a hack to support expressions. We need a better way to
-                            // do this in the future, but certainly not one for this PR.
-                            AccountMethod::DepositBatch(..) => AccountDepositBatchManifestInput {
-                                buckets: Default::default(),
-                            }
-                            .output(node_id, inputs, instruction_index),
-                            AccountMethod::TryDepositBatchOrRefund(..) => {
-                                AccountTryDepositBatchOrRefundManifestInput {
-                                    buckets: Default::default(),
-                                    authorized_depositor_badge: None,
-                                }
-                                .output(node_id, inputs, instruction_index)
-                            }
-                            AccountMethod::TryDepositBatchOrAbort(..) => {
-                                AccountTryDepositBatchOrAbortManifestInput {
-                                    buckets: Default::default(),
-                                    authorized_depositor_badge: None,
-                                }
-                                .output(node_id, inputs, instruction_index)
+                                input.output(details)
                             }
+                            AccountMethod::DepositBatch(input) => input.output(details),
```

Comment:
Ahh right! I wonder why the tests pass then :D.

I suggest we should make `AccountDepositBatchManifestInput = (ManifestValue,)` or _even better_ make a `BucketBatch` which can decode either an Expression::EntireWorktop or a Vec of buckets :D

Reply:
To answer my own questions:

> Ahh right! I wonder why the tests pass then :D.
Because the old behaviour was "no match => no error, just assume unknown resources returned" and none of the tests relied on deposit batch not returning anything.

I've now changed them to error when something can't be decoded.

We still need to:
* Complete defining the packages
* Possibly move this up to the interfaces crate
* Add a test which iterates over every blueprint definition in post-latest DB, and:
  * Ensures Completeness: Checks that resolving a native invocation with `()` results in either a success or a "CantDecode" error.
  * Ensures Schema Accuracy: Checks that resolving a schema for the native invocation aligns with the type from the Database 👍.

> I suggest we should make AccountDepositBatchManifestInput = (ManifestValue,) or even better make a BucketBatch which can decode either an Expression::EntireWorktop or a Vec of buckets :D

I've created `BucketBatch` and `ProofBatch` which turned out to be straightforward 👍.

Reply:
I like the `BucketBatch` approach and think that it's the best way to go about this.

The tests definitely should not pass when the expressions can't be decoded as typed invocations. 

The steps to take you outlined above seem reasonable to me.

I will mark this comment as resolved since the `BucketBatch` type has been implemented.